src/io/iohk/scalanet/peergroup/InetPeerGroupUtils.scala

```diff
@@ -10,7 +10,7 @@ object InetPeerGroupUtils {
 
   type ChannelId = (InetSocketAddress, InetSocketAddress)
 
-  def toTask(f: util.concurrent.Future[_]): Task[Unit] = {
+  def toTask(f: => util.concurrent.Future[_]): Task[Unit] = {
     Task.create[Unit] { (_, cb) =>
       f.addListener(
         (future: util.concurrent.Future[_]) => if (future.isSuccess) cb.onSuccess(()) else cb.onError(future.cause())
```

Comment:
I know this is not related to the PR but, why is this assuming that the future is already completed? if it wasn't successful doesn't mean that it has failed, it may have not completed yet.

Reply:
This is inside listener added by `.addListener` which is called only when the future is completed